File: QE-selenium/java/Activity16.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class Activity16 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/selects");
            System.out.println(driver.getTitle());

          
            WebElement dropdownElement = driver.findElement(By.tagName("select"));
            Select dropdown = new Select(dropdownElement);

            
            dropdown.selectByVisibleText("Two");

           
            dropdown.selectByIndex(3);

        
            dropdown.selectByValue("four");

            
            List<WebElement> options = dropdown.getOptions();
            for (WebElement opt : options) {
                System.out.println(opt.getText());
            }

        } finally {
            driver.quit();
        }
    }
}

File: QE-selenium/python/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import Select

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/selects")
    print(driver.title)
    dropdown = Select(driver.find_element(By.TAG_NAME,"select"))
    dropdown.select_by_visible_text("Two")
    dropdown.select_by_index(3)
    dropdown.select_by_value("four")
    optionss = dropdown.options
    for opt in optionss:
        print(opt.text)
    driver.quit()
